AzureMonitorAgent/agent.py

```diff
@@ -329,6 +328,31 @@ def install():
                 else:  
                     log_and_exit("install", MissingorInvalidParameterErrorCode, 'Parameter "username" and "password" not in proxy protected setting')
 
+        # Determine Managed Identity (MI) settings
+        # Nomenclature: Managed System Identity (MSI), System-Assigned Identity (SAI), User-Assigned Identity (UAI)
+        # Unspecified MI scenario: MSI returns SAI token if exists, otherwise returns UAI token if exactly one UAI exists, otherwise failure
+        # Specified MI scenario: MSI returns token for specified MI
+        if public_settings is not None and "authentication" in public_settings and "managedIdentity" in public_settings.get("authentication"):
+            managedIdentity = public_settings.get("authentication").get("managedIdentity")
+
+            if "identifier-name" not in managedIdentity or "identifier-value" not in managedIdentity:
+                log_and_exit("install", MissingorInvalidParameterErrorCode, 'Parameters "identifier-name" and "identifier-value" are both required in authentication.managedIdentity public setting')
+
+            identifier_name = managedIdentity.get("identifier-name")
+            identifier_value = managedIdentity.get("identifier-value")
+
+            if identifier_name not in ["object_id", "client_id", "mi_res_id"]:
+                log_and_exit("install", MissingorInvalidParameterErrorCode, 'Invalid identifier-name provided; must be "object_id", "client_id", or "mi_res_id"')
+
+            if not identifier_value:
+                log_and_exit("install", MissingorInvalidParameterErrorCode, 'Invalid identifier-value provided; cannot be empty')
+
+            if identifier_name in ["object_id", "client_id"]:
+                guid_re = re.compile(r'[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}')
```

Comment:
In theory I think azure is using UUIDv4, (where 3rd group begins with `4` and 4th group begins with `8`, `9`, `a`, or `b`, but since I don't see any documented proof of this besides my empirical observation, I will stick with the more permissive regex.